File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel


class NotificationRead(BaseModel):
    id: str
    type: str
    channel: str
    payload: dict | None = None
    status: str
    read_at: datetime | None = None
    created_at: datetime


class NotificationListResponse(BaseModel):
    items: list[NotificationRead]
    unread_count: int

File: frontend/src/types/notification.ts
export interface NotificationItem {
  id: string;
  type: string;
  channel: string;
  payload: Record<string, string> | null;
  status: string;
  read_at: string | null;
  created_at: string;
}

export interface NotificationListResponse {
  items: NotificationItem[];
  unread_count: number;
}
